### PR TITLE
[RFC] Release eng process

### PR DESCRIPTION
**Pull request by bkase**
See rfc for details

Thanks @nholland94 for writing the Git Flow section and help planning this document.

### CODE REVIEW

File: rfcs/0020-release-eng-process.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+## Detailed design
+
+[detailed-design]: #detailed-design
+
+### Real Release Branches
+
+This time our "release/beta" branch wasn't really a release branch. In the future, these will be frozen and only critical user-impacting bug fixes will be allowed in. If you do want to make one of these fixes, make a PR against release branch and you'll need to include an explanation as to why you think this PR deserves to land, and you'll need approval from a `release-arbiter`. For now, Brandon will be the sole release arbiter, with the intention of introducing a program to ramp up others and set up a rotation for primary and secondary for a given release.
+
+If the fix for some critical bug is too complicated, the offending commit, if it can be found, will be reverted. If neither of these things can happen, we'll miss our release and this should be considered *bad* and all hands on deck to fix.
```
Comment (imeckler) [inline rfcs/0020-release-eng-process.md]: > this should be considered *bad*

:rofl:

File: rfcs/0020-release-eng-process.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
+## Summary
+[summary]: #summary
+
+Moving to a Git Flow based release process with a hardened release branch and nice release cadence.
```
Comment (psteckler) [inline rfcs/0020-release-eng-process.md]: "nice" isn't very descriptive, is there a better word here? "regular"?

File: rfcs/0020-release-eng-process.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+## Motivation
+
+[motivation]: #motivation
+
+We need a nice release process, and it's best to pick a standard rather than reinventing the wheel because there is nice tooling available and resources online to get help.
```
Comment (psteckler) [inline rfcs/0020-release-eng-process.md]: again

File: rfcs/0020-release-eng-process.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+### Schedule
+
+For now, we'll plan on releasing every Monday @ 2pm, starting on 8/5
+
+#### Next week:
+
+- Monday day, move `master -> develop` and `stable -> master`
+    - This also entails fixing whatever tooling we have in place to point to the new branches, and updating the configuration of our documentation so the edit-button points to develop branch
+- Cut the new `release/*` branch on EOD Monday, and have a testnet candidate online by Tuesday
```
Comment (psteckler) [inline rfcs/0020-release-eng-process.md]: Is `release` from `master`?

File: rfcs/0020-release-eng-process.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+During the period in which the release branch is open, any bugs assigned to you should be considered very high priority.
+
+### Schedule
+
+For now, we'll plan on releasing every Monday @ 2pm, starting on 8/5
+
+#### Next week:
+
+- Monday day, move `master -> develop` and `stable -> master`
```
Comment (psteckler) [inline rfcs/0020-release-eng-process.md]: I don't see other mentions of `stable` ... where does that fit with other branches?

Comment (bkase) [reply rfcs/0020-release-eng-process.md]: Stable is a temporary hack that we aliased what formerly was `release/beta`

**Comment by imeckler** [comment]
Pursuant to some discussion this morning, what do people think of Wednesday code freeze, Tuesday release. If that's enough time, we'll push it to Thursday code freeze.

**Comment by bkase** [comment]
@ihm I’d like to still freeze it tonight/tomorrow morning and use the extra day if necessary this week. Next week we can try Wednesday freeze if it goes well this time.

**Comment by yourbuddyconner** [comment]
You don't include anything in here specifically about how/when branches will be tagged. This is important for automation.

**Pull request closed without merging**